octodns/provider/easydns.py

```diff
@@ -0,0 +1,453 @@
+        # EasyDNS allows for new domains to be created for the purpose of DNS
+        # only, or with domain registration. This function creates a DNS only
+        # record expectig the domain to be registered already
+        path = '/domains/add/{}'.format(name)
+        domainData = {'service': 'dns',
+                      'term': 1,
+                      'dns_only': 1,
+                      'portfolio': self.domainPortfolio,
+                      'currency': self.defaultCurrency}
+        self._request('PUT', path, data=domainData).json()
+
+        # EasyDNS creates default records for MX, A and CNAME for new domains,
+        # we need to delete those default record so we can sync with the source
+        # records, first we'll sleep for a second before gathering new records
+        # We also create default NS records, but they won't be deleted
+        sleep(1)
+        records = self.records(name, True)
+        for record in records:
+            if record['type'] in ('A', 'MX', 'CNAME'):
+                self.record_delete(name, record['id'])
+
+    def records(self, zone_name, raw=False):
+        if raw:
+            path = '/zones/records/all/{}'.format(zone_name)
+        else:
+            path = '/zones/records/parsed/{}'.format(zone_name)
+
+        ret = []
+        resp = self._request('GET', path).json()
+        ret += resp['data']
+
+        # EasyDNS supports URL forwarding, stealth URL forwarding and DYNamic
+        # A records so we'll convert them to their underlying DNS record
+        # types before processing
+        for record in ret:
+            # change any apex record to empty string
+            if record['host'] == '@':
+                record['host'] = ''
+
+            # change any apex value to zone name
+            if record['rdata'] == '@':
+                record['rdata'] = '{}.'.format(zone_name)
+
+            # change "URL" & "STEALTH" to a "CNAME"
+            if record['type'] == "URL" or record['type'] == "STEALTH":
+                record['type'] = 'CNAME'
+
+            if record['type'] == "DYN":
+                record['type'] = 'A'
+
+        return ret
+
+    def record_create(self, zone_name, params):
+        path = '/zones/records/add/{}/{}'.format(zone_name, params['type'])
+        # change empty name string to @, EasyDNS uses @ for apex record names
+        params['host'] = params['name']
+        if params['host'] == '':
+            params['host'] = '@'
+        self._request('PUT', path, data=params)
+
+    def record_delete(self, zone_name, record_id):
+        path = '/zones/records/{}/{}'.format(zone_name, record_id)
+        self._request('DELETE', path)
+
+
+class EasyDNSProvider(BaseProvider):
+    '''
+    EasyDNS provider using API v3
+
+    easydns:
+        class: octodns.provider.easydns.EasyDNSProvider
+        # Your EasyDNS API token (required)
+        token: foo
+        # Your EasyDNS API Key (required)
+        apikey: bar
```

Comment:
apikey would likely be more consistent with other providers as `api_key`, but can stay this way if easy dns refers to it as a single word/combined like that.

Reply:
No problem, happy to keep it consistent.